jsystemd-spring-boot-starter/src/main/java/com/github/jpmsilva/jsystemd/SystemdSpringApplicationRunListener.java

```diff
@@ -30,7 +33,11 @@
  * @see SpringApplicationRunListener
  */
 public class SystemdSpringApplicationRunListener implements SpringApplicationRunListener {
+  private static final String SYSTEMD_BEAN_NAME = "systemd";
+  private static final boolean isUnderSystemd = SystemdUtilities.isUnderSystemd();
+  private static Systemd systemd;
```

Comment:
Could you refactor ensureSystemd to return the created systemd object (or null) and call it here from the static initializer context?
That way you can remove the synchronized qualifier from the method.

Reply:
Done as per your suggestion